core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/UpgradePermissionProvider.java

```diff
@@ -23,12 +23,17 @@
     private ComponentVersionComparator componentVersionComparator;
 
     public boolean permitCmAndStackUpgrade(ImageFilterParams imageFilterParams, Image image, String versionKey, String buildNumberKey) {
+        return permitCmAndStackUpgrade(imageFilterParams, image, versionKey, buildNumberKey, imageFilterParams.isCheckUpgradeMatrix());
```

Comment:
Wouldn't it be better to set the imageFilterParams  checkUpgradeMatrix to false in case of CM instead of creating a different method where you can override it. It will be difficult to maintain later.

Reply:
I agree that it may not be the best solution, but overriding the imageFilterParams for one call does not seems better. How about separating upgradePermissionProvider.permitCmAndStackUpgrade to 2 different methods? And in the cm method we do not care about the matrix.

Reply:
Yeah, that seems better.